Readme Update

MitchellShiell opened the pull request:
Re-posting this PR, all the other main Overture readmes are drafted. I'm holding back on pulling them before getting feedback on this one. As discussed offline, I plan on cutting down the pre-amble/marketing and bringing the development setup closer to the top.

Commit:
readmeupdate

MitchellShiell committed:
Update README.md

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
-## Documentation
-
-Explore documentation with the Ego [Read the Docs](https://ego.readthedocs.io/en/latest/).
+<!-- Replace slack with discourse once setup -->
 
-## Introduction
+[<img src="https://img.shields.io/badge/chat-on--slack-blue">](http://slack.overture.bio) 
+[<img src="https://img.shields.io/badge/License-gpl--v3.0-blue">](https://github.com/overture-stack/ego/blob/develop/LICENSE)
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+## Worry Less Science More
```
This Repo is for Ego, not Overture. The most prominent entry on the top line should be the name Ego and any branding associated with it, not the orgnaization that is producing the application. The previous readme started with an `h1` for Ego, where more traditionally it would use a markdown title like `# Ego`.

Overture branding can follow but the first header/title should be about the application stored in this repo.

MitchellShiell replied:
Addressed!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+## Worry Less Science More
 
-Authorization Service built to provide Single Sign On for various microservices in an application. EGO works with Identity Providers such as Google, Facebook to provide social logins in the application. EGO provides stateless authorization using [JWT (JSON Web Tokens)](https://jwt.io/) and can scale very well to a large number of users.
+In this Overture repository, we (the [OICR Genome Informatics Team](https://softeng.oicr.on.ca/team/)) develop [Ego](https://www.overture.bio/products/ego/), our open-source authentication and authorization microservice. [Overture](https://www.overture.bio/) is an ensemble of modular solutions for big-data genomic science. Our core products work in concert to manage, explore and visualize molecul ...
```
No need to declare `Overture repository`, we already have github indicating that `overture-stack` is the organization this repository belongs in.

The first paragraph should focus on details of the ego application and not the developers.

I would rearrange the first two sections you have to put the Ego section at the top, then move the Overture section beneath. The overture section perhaps could have a standard re-usable paragraph about what overture is, then a custom paragraph about how Ego fits into the Overture application suite.

Something like:


# Ego

## Secure Authentication & Authorization
Ego is an open-source authentication and authorization microservice, providing secure user registration and permission management. Ego uses well-known single-sign-on identity providers like Google, GitHub, LinkedIn and ORCiD in place of managing usernames and passwords. 

## Overture - Worry Less, Science More
< overutre logo here >
[Overture](https://www.overture.bio/) is an ensemble of modular solutions for big-data genomics science. Our core products work in concert to manage, explore and visualize molecular and clinical data. Visit our [website](https://www.overture.bio/) for more information on what Overture offers, and check out our other projects on [GitHub](https://github.com/overture-stack/).

MitchellShiell replied:
Addressed!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
-- [OpenID Connect](http://openid.net/connect/)
+|Product|Description|
+|---|---|
+|[Ego](https://www.overture.bio/products/ego/)|A stateless authorization and user management service|
```
Ego provides statless (JWT) and stateful (API-Key) authorization now, so we should probably drop the `stateless` descriptor. The rest is good.

MitchellShiell replied:
got it!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+<p align="left"><img alt="Overture overview" src="https://www.overture.bio/static/124ca0fede460933c64fe4e50465b235/a6d66/system-diagram.png"></p>
 
-## Tech Stack
+Overtures' modular architecture allows you to utilize and mix any of our products to fulfill your individual needs. Our core technologies, however, can also work in concert as an end-to-end data management system (DMS) designed to satisfy the needs of modern large-scale genomic research. For more information on our DMS, please see our [DMS documentation](https://www.overture.bio/documentation/dms ...
```
Overture's

joneubank replied:
I would start with

> Ego is a central component in the Overture Data Management System. < link to DMS >

then continue as is you have (slight text modifications).

> Overture's modular architecture allows you to utilize and mix any of our products to fulfill your individual needs. Our core technologies can be used as standalone services, and can alsobe deployed in concert as an end-to-end data management system (DMS) designed to satisfy the needs of modern large-scale genomics research.

joneubank replied:
Note: I'm pluralizing `genomics` by reflex, I'm not sure if `genomic research` or `genomics research` or both are correct

MitchellShiell replied:
Got it!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+|---|---|
+|[Ego](https://www.overture.bio/products/ego/)|A stateless authorization and user management service|
+|[Score](https://www.overture.bio/products/score/)| Transfer data quickly and easily to and from any cloud-based storage system|
+|[Song](https://www.overture.bio/products/song/)|Quickly and reliably track genome metadata scattered across multiple Cloud storage systems|
```
Being nitpicky here, but the central role of song is to track file metadata, and optionally add in genomics data.

Perhaps:

> Catalog and manage metadata of genomics data spread across cloud storage systems

joneubank replied:
Also, we need consistency on if we capitalize `Cloud`, I vote we let it be lower case

MitchellShiell replied:
Got it!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+|[Ego](https://www.overture.bio/products/ego/)|A stateless authorization and user management service|
+|[Score](https://www.overture.bio/products/score/)| Transfer data quickly and easily to and from any cloud-based storage system|
+|[Song](https://www.overture.bio/products/song/)|Quickly and reliably track genome metadata scattered across multiple Cloud storage systems|
+|[Maestro](https://www.overture.bio/products/maestro/)|Organizing your distributed data into one index|
```
suggestion:
> into a centralized Elasticsearch index

MitchellShiell replied:
Got it!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+Ego is [OAuth 2.0](https://oauth.net/2/) and [OpenID Connect](https://auth0.com/docs/authenticate/protocols/openid-connect-protocol) compliant. It is written in JAVA and uses [Sprint Boot](https://spring.io/projects/spring-boot) and [Spring Security Frameworks](https://spring.io/projects/spring-security). Because it uses [JSON Web Tokens (JWT)](https://jwt.io/) for stateless authorization, it can ...
 
-Here are some of the features of EGO:
+## Related Products 
```
Can we add a section in Related Products about Ego UI (and move the screen shot there?)

MitchellShiell replied:
Got it!

joneubank commented on README.md:
```diff
@@ -1,239 +1,126 @@
+<!--Requirements Post-gres & JDK?-->
 
-#### Database Migrations with Flyway
+This is a step-by-step guide for setting up a dockerized version of Ego. See our [setup section](#setup) below for a comprehensive setup guide.
```
> of Ego and Ego UI

MitchellShiell replied:
Got it!

joneubank reviewed (changes requested):
Great update, a lot of content thats great to see. The comments are just a bunch of my opinions on text changes and some nit picking on technical details.

My main request is to make the focus of the readme about Ego, and move the Overture branding and discussion to the second section.

We also removed a bit of text from the readme that we should get captured elsewhere in this repo, so we should also open some tickets to track needed documentation changes, just leaving this list as reminders for us to make those issues:
1. docs on running ego as a developer
2. docs on ego configuration and run profiles
3. readme section on ego technologies and dependencies

Beyond that we can improve our documentation with explanations of the ego data model (users, applications, groups, permissions, api-keys) and how to interact with it.

MitchellShiell committed:
Update README.md

MitchellShiell committed:
Delete Overture_logo.png

MitchellShiell committed:
Add files via upload

MitchellShiell commented:
Draft 2 of Readme Update

Thanks for all the feedback, happy with the progress on this have a look and leave your thoughts.

Elements should line up correctly when viewing from repo
https://github.com/MitchellShiell/ego/tree/readmeupdate

Minimized overture branding and focused on Ego

Restructured content and slimmed down the writing for better readability

Updated badges and images

I'll reiterate what you mentioned previously; An update to the Github Wiki is required:

1. Running ego as a developer

2. Ego configuration and run profiles

3. Explanations of the ego data model (users, applications, groups, permissions, api-keys) and how to interact with it.

Beyond that adding on technical specifications and dependencies required within the technical specifications section

MitchellShiell committed:
Docker Pulls Removed

As discussed offline, removed Docker Pull, No GHRC metrics with BadgeIO currently

joneubank commented on README.md:
```diff
@@ -1,239 +1,108 @@
 
-When run locally this can be found at: [http://localhost:8081/swagger-ui.html](http://localhost:8081/swagger-ui.html)
+Biomedical data requires secure protocols for authenticating users and authorizing the information and applications those users can access. [Ego](https://www.overture.bio/products/ego/) addresses this by facilitating user registration and providing a secure permission management system. 
+An [Ego UI tool](https://github.com/overture-stack/ego-ui) was also developed to make these services accessible to all collaborators.
```
don't need "tool", just `Ego UI` should be fine. Rephrasing like:
> Ego UI is also available to make the Ego service accessible for system administrators.

MitchellShiell replied:
removed!

joneubank commented on README.md:
```diff
@@ -1,239 +1,108 @@
+Overture is an ecosystem of research software tools, each with narrow responsibilities, designed to address the adapting needs of genomics research. 
 
-To register an application with EGO make a request as documented at [/swagger-ui.html#!/application-controller/createUsingPOST](/swagger-ui.html#!/application-controller/createUsingPOST)
+The DMS is a fully functional and customizable data portal built from a packaged collection of Overtures microservices. For more information on our DMS, read our [DMS documentation](https://www.overture.bio/documentation/dms/).
```
```suggestion
The Overture **Data Management System** (DMS) is a fully functional and customizable data portal built from a packaged collection of Overture's microservices. For more information on DMS, read our [DMS documentation](https://www.overture.bio/documentation/dms/).
```

joneubank reviewed (approved):
Amazing refresh. Very impressive look now. Thank you!
Added two comments with small phrasing suggestions and one typo found, consider updating before merging.

MitchellShiell committed:
Update README.md

Co-authored-by: Jon Eubank <joneubank@gmail.com>

MitchellShiell committed:
Update README.md

MitchellShiell committed:
Merge branch 'readmeupdate' of https://github.com/MitchellShiell/ego into readmeupdate
